lib/jsonWebToken.js

```diff
@@ -0,0 +1,186 @@
+  if (!callback)
+    throw new Error('callback must be provided in jsonWebToken.verify');
+  if (typeof callback !== 'function')
+    throw new Error('callback in jsonWebToken.verify must be a function');
+
+  // asynchronous wrapper for callback
+  var done = function() {
+    var args = Array.prototype.slice.call(arguments, 0);
+    return process.nextTick(function() {
+      callback.apply(null, args);
+    });
+  };
+
+  // make sure we have the required fields in options
+  if (!options.audience)
+    return done(new Error('options.audience is missing in jsonWebToken.verify'));
+  if (!options.algorithms)
+    return done(new Error('options.algorithms is missing in jsonWebToken.verify'));
+  if (!Array.isArray(options.algorithms) || options.algorithms.length == 0)
+    return done(new Error('options.algorithms must be an array containing at least 1 algorithm'));
+
+  /*********************************************************************
+   * Checking jwtString structure, getting header, payload and signature
+   ********************************************************************/
+
+  // split jwtString, make sure we have three parts and we have signature
+  var parts = jwtString.split('.');
+  if (parts.length !== 3)
+    return done(new Error('jwtString is malformed'));
+  if (parts[2] === '')
+    return done(new Error('signature is missing in jwtString'));
+  
+  // decode jwsString
+  var decodedToken;
+  try {
+    decodedToken = jws.decode(jwtString);
+  } catch(err) {
+    return done(new Error('invalid token'));
+  }
+
+  if (!decodedToken) {
+    return done(new Error('invalid token'));
+  }
+
+  // get header, payload and signature
+  var header = decodedToken.header;
+  var payload = decodedToken.payload;
+  var signature = decodedToken.signature;
+
+  if (!header)
+    return done(new Error('missing header in the token'));
+  if (!payload)
+    return done(new Error('missing payload in the token'));
+  if (!signature)
+    return done(new Error('missing signature in the token'));
+
+  /*********************************************************************
+   * validate algorithm and signature
+   ********************************************************************/
+
+  // alg in the token header should be one of those provided in options.algorithms
+  if (!~options.algorithms.indexOf(header.alg)) {
+    return done(new Error('invalid algorithm'));
+  }
+
+  try {
+    var valid = jws.verify(jwtString, header.alg, PMEKey);
+    if (!valid)
+      return done(new Error('invalid signature'));
+  } catch (e) {
+    return done(e);
+  }
+
+  /*********************************************************************
+   * validate payload content
+   ********************************************************************/
+
+  // (1) issuer
+  //   - check the existence and the format of payload.iss
+  //   - validate if options.issuer is set
+  if (typeof payload.iss !== 'string')
```

Comment:
This code will need to handle multiple issuers being handed in to support issuer validation on the common endpoint. options.issuer may need to be able to be an array.